pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -1040,16 +1040,24 @@ func containerMemoryFromBlock(blocks []memorymanagerstate.Block) []*podresources
 	return containerMemories
 }
 
-func (cm *containerManagerImpl) PrepareResources(pod *v1.Pod, container *v1.Container) (*dra.ContainerInfo, error) {
-	return cm.draManager.PrepareResources(pod, container)
+func (cm *containerManagerImpl) PrepareDynamicResources(pod *v1.Pod) error {
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DynamicResourceAllocation) {
+		return cm.draManager.PrepareResources(pod)
+	}
+
+	return nil
 }
 
-func (cm *containerManagerImpl) UnprepareResources(pod *v1.Pod) error {
-	return cm.draManager.UnprepareResources(pod)
+func (cm *containerManagerImpl) UnprepareDynamicResources(pod *v1.Pod) error {
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DynamicResourceAllocation) {
```

Comment:
We should remove the check here and instead put it around callers (as appropriate)